source/jormungandr/jormungandr/realtime_schedule/cleverage.py

```diff
@@ -0,0 +1,147 @@
+        self.service_url = service_url if (service_url[-1] == u'/') else (service_url+'/')
+        self.service_args = service_args
+        self.timeout = timeout  # timeout in seconds
+        self.rt_system_id = id
+        self.object_id_tag = object_id_tag if object_id_tag else id
+        self.breaker = pybreaker.CircuitBreaker(fail_max=app.config['CIRCUIT_BREAKER_MAX_CLEVERAGE_FAIL'],
+                                                reset_timeout=app.config['CIRCUIT_BREAKER_CLEVERAGE_TIMEOUT_S'])
+        self.timezone = pytz.timezone(timezone)
+
+    def __repr__(self):
+        """
+         used as the cache key. we use the rt_system_id to share the cache between servers in production
+        """
+        return self.rt_system_id
+
+    @cache.memoize(app.config['CACHE_CONFIGURATION'].get('TIMEOUT_CLEVERAGE', 30))
+    def _call_cleverage(self, url):
+        """
+        http call to cleverage
+        """
+        try:
+            return self.breaker.call(requests.get, url, timeout=self.timeout, headers=self.service_args)
+        except pybreaker.CircuitBreakerError as e:
+            logging.getLogger(__name__).error('Cleverage RT service dead, using base '
+                                              'schedule (error: {}'.format(e))
+        except requests.Timeout as t:
+            logging.getLogger(__name__).error('Cleverage RT service timeout, using base '
+                                              'schedule (error: {}'.format(t))
+        except:
+            logging.getLogger(__name__).exception('Cleverage RT error, using base schedule')
+        return None
+
+    def _make_url(self, route_point):
+        """
+        The url returns something like a departure on a stop point
+        """
+
+        stop_id = route_point.fetch_stop_id(self.object_id_tag)
+
+        if not stop_id:
+            # one a the id is missing, we'll not find any realtime
+            logging.getLogger(__name__).debug('missing realtime id for {obj}: stop code={s}'.
+                                              format(obj=route_point, s=stop_id))
+            return None
+
+        url = "{base_url}{stop_id}".format(base_url=self.service_url, stop_id=stop_id)
+
+        return url
+
+    def _get_dt(self, datetime_str):
+        #"2016-04-11 14:37:15"
+        dt = datetime.strptime(datetime_str, "%Y-%m-%d %H:%M:%S")
+
+        utc_dt = self.timezone.normalize(self.timezone.localize(dt)).astimezone(pytz.utc)
+
+        return utc_dt
+
+    def _get_passages(self, route_point, cleverage_resp):
+        logging.getLogger(__name__).debug('cleverage response: {}'.format(cleverage_resp))
+
+        line_code = route_point.fetch_line_code()
+
+        schedules = [line['schedules'] for line in cleverage_resp if line['code'].lower() == line_code.lower()]
+
+        if schedules:
+            next_passages = []
+            for next_expected_st in schedules[0]:
```

Comment:
why do you consider only the first line ? there can't be many lines ? it that's the case, maybe we should put a warning when it's the case no ?